rust/rope/src/diff.rs

```diff
@@ -250,43 +245,19 @@ impl DiffBuilder {
     }
 }
 
-/// Fast iterator over lines in a string, not removing newline characters.
-struct LineScanner<'a> {
-    inner: &'a str,
-    idx: usize,
-}
-
-impl<'a> Iterator for LineScanner<'a> {
-    type Item = &'a str;
-    fn next(&mut self) -> Option<&'a str> {
-        if self.idx >= self.inner.len() {
-            return None;
-        }
-
-        match memchr(b'\n', &self.inner.as_bytes()[self.idx..]) {
-            Some(idx) => {
-                let next_idx = self.idx + idx + 1;
-                let result = &self.inner[self.idx..next_idx];
-                self.idx = next_idx;
-                Some(result)
-            }
-            None => {
-                let result = &self.inner[self.idx..];
-                self.idx = self.inner.len();
-                Some(result)
-            }
-        }
-    }
-}
-
-fn make_line_hashes<'a>(base: &'a str, min_size: usize) -> HashMap<&'a str, usize> {
+/// Create a HashMap based on each line in the rope wich respect a mininum line length.
```

Comment:
I appreciate adding the comment! I might phrase it slightly different, though, how about:

>Creates a map of lines to offsets, ignoring trailing whitespace, and only for those lines where `line.len() >= min_size`. Offsets refer to the first non-whitespace byte in the line.

Reply:
Yes, it's done.